app/gui/src/dashboard/layouts/AssetContextMenu.tsx

```diff
@@ -91,16 +86,10 @@ export default function AssetContextMenu(props: AssetContextMenuProps) {
   const downloadAssetsMutation = reactQuery.useMutation(downloadAssetsMutationOptions(backend))
   const self = permissions.tryFindSelfPermission(user, asset.permissions)
   const isCloud = categoryModule.isCloudCategory(category)
-  const pathComputed =
-    category.type === 'recent' || category.type === 'trash' ? null
-    : isCloud ? computeFullRemotePath(asset, users, userGroups)
-    : asset.type === backendModule.AssetType.project ?
-      mapNonNullish(localBackend?.getProjectPath(asset.id) ?? null, normalizePath)
-    : normalizePath(localBackendModule.extractTypeAndId(asset.id).id)
   const path =
-    pathComputed == null ? null
-    : isCloud ? encodeURI(pathComputed)
-    : pathComputed
+    asset.ensoPath == null ? null
+    : isCloud ? encodeURI(asset.ensoPath)
```

Comment:
Can we do that in the RemoteBackend? I mean, the `isCloud` seems dirty...

Reply:
true! i don't know tbh. while i'm not *against* it, think there might be some situations where we might want the original, non-escaped path - e.g. for display:

https://github.com/enso-org/enso/blob/8227747e04b47764ecdd72e437d9c5196c1f073b/app/gui/src/dashboard/layouts/AssetPanel/components/AssetProperties.tsx#L275-L278

although... honestly i can see what i can do

Reply:
might be a good idea to ask pawel or james about whether this is actually necessary though given that the cloud now does return something that should theoretically be the "proper" URL?